tests/plugins/colordialog/_helpers/tools.js

```diff
@@ -0,0 +1,40 @@
+/* exported assertColor, assertColorAtDialogShow */
+
+function assertColor( editor, inputColor, outputColor ) {
+	editor.once( 'dialogShow', function( evt ) {
+		var dialog = evt.data;
+		dialog.setValueOf( 'picker', 'selectedColor', inputColor );
+		dialog.getButton( 'ok' ).click();
+	} );
+
+	editor.getColorFromDialog( function( color ) {
+		resume( function() {
+			assert.areSame( outputColor, color );
+		} );
+	} );
+	wait();
+}
+
+function assertColorAtDialogShow( editor, expectedColor, html, button ) {
+	var toolbarButton = editor.ui.get( button );
+	editor.insertHtml( html );
+	editor.once( 'dialogShow', function( evt ) {
+		resume( function() {
+			var dialog = evt.data,
+				selectedColor = dialog.getValueOf( 'picker', 'selectedColor' );
+			dialog.getButton( 'ok' ).click();
+
+			assert.areSame( expectedColor, selectedColor );
+		} );
+	} );
+
+	toolbarButton.click( editor );
```

Comment:
IMO the more elegant way is to invoke appropriate editor's command, instead of simulating click on toolbar's button.

Reply:
Some logic is nested inside functions which are not exposed outside plugin. To use editor's command `editor.getColorFromDialog()`I would have to either include significant part of plugin code in tests or explicitly pass the color that should be set (and then checked) in the dialog. Therefore I chose to simulate button click, which seems to be the best way to test if feature works correctly.